src/main/java/io/cdap/plugin/fileappend/FileAppenderSinkConfig.java

```diff
@@ -0,0 +1,228 @@
+package io.cdap.plugin.fileappend;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
+
+import java.io.IOException;
+import javax.annotation.Nullable;
+
+/**
+ * Config properties for the plugin.
+ */
+public class FileAppenderSinkConfig extends PluginConfig {
+  public static final String SCHEMA = "schema";
+  public static final String SIZE_THRESHOLD = "sizeThreshold";
+  public static final String AGE_THRESHOLD = "ageThreshold";
+
+  @Description("The name of the FileSet to write to.")
+  private final String name;
+
+  @Nullable
+  @Description("Optional directory within the FileSet to write to.")
+  private final String outputDir;
+
+  @Description("The prefix to use for all files in the FileSet.")
+  private final String fileNamePrefix;
+
+  @Nullable
+  @Description("Rotate files after they are this large (in mb). 0 means no limit. Defaults to 100.")
+  private final Long sizeThreshold;
+
+  @Nullable
+  @Description("Rotate files after they are this old (in minutes). 0 means no limit. Defaults to 60.")
+  private final Long ageThreshold;
+
+  @Nullable
+  @Description("The separator to use to join input record fields together. Defaults to ','.")
+  private final String fieldSeparator;
+
+  @Nullable
+  @Description("The schema of the FileSet.")
+  private final String schema;
+
+  public FileAppenderSinkConfig(
+    String name,
+    @Nullable String outputDir,
+    String fileNamePrefix,
+    @Nullable Long sizeThreshold,
+    @Nullable Long ageThreshold,
+    @Nullable String fieldSeparator,
+    @Nullable String schema
+  ) {
+    this.name = name;
+    this.outputDir = outputDir;
+    this.fileNamePrefix = fileNamePrefix;
+    this.sizeThreshold = sizeThreshold;
+    this.ageThreshold = ageThreshold;
+    this.fieldSeparator = fieldSeparator;
+    this.schema = schema;
+  }
+
+  private FileAppenderSinkConfig(Builder builder) {
+    name = builder.name;
+    outputDir = builder.outputDir;
+    fileNamePrefix = builder.fileNamePrefix;
+    sizeThreshold = builder.sizeThreshold;
+    ageThreshold = builder.ageThreshold;
+    fieldSeparator = builder.fieldSeparator;
+    schema = builder.schema;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(FileAppenderSinkConfig copy) {
+    return new Builder()
+      .setName(copy.getName())
+      .setOutputDir(copy.getOutputDir())
```

Comment:
what would happen when output directory is null or empty? since its an optional field

Reply:
similar question for other fields as well.